module/input/SensorFilter/src/SensorFilter.cpp

```diff
@@ -374,16 +374,14 @@ namespace module::input {
         // Convert the rotation matrix from the mahony method into euler angles
         Eigen::Vector3d rpy_mahony = mat_to_rpy_intrinsic(Rwt_mahony);
         // Remove yaw from mahony filter (prevents it breaking after numerous rotations)
-        mahony_filter.set_state(rpy_intrinsic_to_mat(Eigen::Vector3d(rpy_mahony.x(), rpy_mahony.y(), 0.0)));
+        // mahony_filter.set_state(rpy_intrinsic_to_mat(Eigen::Vector3d(rpy_mahony.x(), rpy_mahony.y(), 0.0)));
```

Comment:
commented out code

Reply:
This part is the Mahony yaw being used again, I assume your tuning got it working? It does look like it should be another PR. Has this been tested on the real robot?